treeherder/seta/analyze_failures.py

```diff
@@ -0,0 +1,141 @@
+
+        For instance, in the raw data you might see a reference to 9fa614d8310d which is a back out
+        and it is reference by 12 starred jobs:
+            https://treeherder.mozilla.org/#/jobs?repo=autoland&filter-searchStr=android%20debug%20cpp&tochange=9fa614d8310db9aabe85cc3c3cff6281fe1edb0c
+        The raw data will show those 12 jobs.
+
+        We return the obtained data or an empty structure if we failed to fetch it.
+
+        [1]
+        {
+          "failures": {
+            "44d29bac3654": [
+              ["android-4-0-armv7-api15", "opt", "android-lint", 2804 ],
+              ["android-4-0-armv7-api15", "opt", "android-api-15-gradle-dependencies", 2801],
+        """
+        # XXX: Once this uses data from Treeherder we will be able to query the table directly
+        url = self.seta_host + "/data/seta/?startDate=%s&endDate=%s" % (
+            self.start_date, self.end_date)
+        LOG.info('Grabbing information from {}'.format(url))
+        try:
+            response = retry(requests.get,
+                             args=(url, ),
+                             kwargs={
+                                 'headers': {
+                                     'Accept': 'application/json',
+                                     'User-Agent': 'treeherder',
+                                 },
+                                 'verify': True
+                             })
+            data = json.loads(response.content)
+        except Exception as error:
+            # XXX: Should we not simply abort the script?
+            # we will return an empty 'failures' list if got exception here
+            LOG.warning("The request for %s failed due to %s" % (url, error))
+            LOG.warning("We will work with an empty list of failures")
+            data = {'failures': []}
+
+        LOG.info("start-date: {}, end-date: {}, failures: {}".format(
+             self.start_date, self.end_date, len(data['failures'])))
+        return data['failures']
+
+    def clear_expiration_field_for_expired_jobs(self):
+        # Only select rows where there is an expiration date set
+        data = JobPriority.objects.filter(expiration_date__isnull=False)
+
+        for job in data:
+            if job.has_expired():
+                LOG.info('Clearing expiration date for {}'.format(job))
+                job.expiration_date = None
+                job.save()
+
+    def increase_jobs_priority(self, high_value_jobs, priority=1, timeout=0):
+        """For every high value job  see if we need to adjust the priority in the database
+
+        Currently, high value jobs have a priority of 1 and a timeout of 0.
+
+        Return how many jobs had their priority increased
+        """
+        LOG.info("Let's see if we need to increase the priority of any job")
+        # NOTE: We ignore job priorities with expiration date set
+        original_queryset = JobPriority.objects.filter(expiration_date__isnull=True)
```

Comment:
How about `job_priorities_with_expirations`? `original_queryset` is a bit ambiguous.